tests/app/main/test_suppliers.py

```diff
@@ -80,7 +80,9 @@ def get_supplier(*args, **kwargs):
         }
     }
     supplier_info.update(kwargs)
-    return {"suppliers": supplier_info}
+    return {"suppliers": dict(
+        filter(lambda x: x[1] is not None, supplier_info.items())
```

Comment:
what does this filtering do? :)

Reply:
Given a dictionary such as `{'a': True, 'b': False, 'c': None}` it will filter out any key/value pairs where the value is not None, i.e. you'll be left with ``{'a': True, 'b': False}`.

`supplier_info.items()` groups the key/value pairs into tuples like so: `[('a', True), ('b', False), ('c', None)]`. The filter then takes each tuple and checks the second item (and index 1) against `is not None`.

Reply:
Ahh, makes a lot of sense,  thanks a lot! :)

Reply:
The reason I did this is because when the API serialises a supplier, it purges all keys with a null value.

Adding this filter in should make our tests more realistic

Reply:
Ahh, smart move! :)